js/src/Ice/CommunicatorExtensions.js

```diff
@@ -0,0 +1,14 @@
+
+import { Communicator } from './Communicator.js';
+import { Instance } from './Instance.js';
+import { AsyncResultBase } from './AsyncResultBase.js';
+
+Communicator.prototype.createInstance = function(initData)
```

Comment:
This methods are defined in CommunicatorExtensions to avoid circular dependencies but are still part of the Communicator class.